src/generic_core/core.ts

```diff
@@ -435,44 +435,76 @@ class uProxyCore implements uProxy.CoreAPI {
+    'https://www.uproxy.org/submit-feedback'
+  ]
+
+  private post_ = (url :string, data :Object) :Promise<void> => {
     return new Promise<void>((fulfill, reject) => {
-      var sendXhr = (logs) : void => {
-        var xhr = freedom["core.xhr"]();
-        xhr.on('onreadystatechange', () => {
-          Promise.all([xhr.getReadyState(), xhr.getStatus()])
-            .then((stateAndStatus) => {
-              // 200 is the HTTP result code for a successful request.
-              if (stateAndStatus[0] === XMLHttpRequest.DONE && stateAndStatus[1] === 200) {
-                fulfill();
-              } else if (stateAndStatus[0] === XMLHttpRequest.DONE && stateAndStatus[1] != 200) {
-                // TODO: Once we have non-AppEngine links we can send feedback to, try
-                // multiple URLs before rejecting the sendFeedback promise.
-                // https://github.com/uProxy/uproxy/issues/1191
-                reject('POST to uproxy.org failed.');
-              }
-            });
+      var xhr = freedom['core.xhr']();
+
+      xhr.on('onreadystatechange', () => {
+        Promise.all([xhr.getReadyState(), xhr.getStatus()])
+        .then((stateAndStatus) => {
+          // 200 is the HTTP result code for a successful request.
+          if (stateAndStatus[0] === XMLHttpRequest.DONE) {
+            if (stateAndStatus[1] === 200) {
+              fulfill();
+            } else {
+              reject(new Error('POST failed with HTTP code ' + stateAndStatus[1]));
+            }
+          }
         });
-        var params = JSON.stringify(
-          {'email' : feedback.email,
-           'feedback' : feedback.feedback,
-            'logs' : logs});
-        xhr.open('POST', 'https://beta-dot-uproxysite.appspot.com/submit-feedback', true);
-        // core.xhr requires the parameters to be tagged as either a
-        // string or array buffer in the format below.
-        // This is roughly equivalent to standard xhr.send(params).
-        xhr.send({'string': params});
-      }
+      });
+      var params = JSON.stringify(data);
 
-      var browserInfo = 'Browser Info: ' + feedback.browserInfo + '\n\n';
+      xhr.open('POST', url, true);
+      // core.xhr requires the parameters to be tagged as either a
+      // string or array buffer in the format below.
+      // This is roughly equivalent to standard xhr.send(params).
+      xhr.send({'string': params});
+    });
+  }
 
-      if (feedback.logs) {
-        this.getLogsAndNetworkInfo().then((logsWithNetworkInfo) => {
-          sendXhr(browserInfo + logsWithNetworkInfo);
-        });
-      } else {
-        sendXhr('');
+  public sendFeedback = (feedback :uProxy.UserFeedback, maxAttempts?:number) : Promise<void> => {
+    if (!maxAttempts || maxAttempts > this.FeedbackUrls_.length) {
+      // default to trying every possible URL
+      maxAttempts = this.FeedbackUrls_.length;
+    }
+
+    var logs :Promise<string>;
```

Comment:
Would you be fine with just `getLogs`?  I don't like the idea of including typing in variable names.